tests/run/fully-abstract-interface.check

```diff
@@ -0,0 +1,29 @@
+CaseClassImplementation
+underlying rep: class CaseClassImplementation$Const
+1
+test1 OK
+1 = 1
+test2 OK
+1 = 1
+
+underlying rep: class CaseClassImplementation$App
+7
+test3 OK
+AppliedOp(PlusOp class CaseClassImplementation$Const class CaseClassImplementation$App) = 7
+test4 OK
+AppliedOp(PlusOp class CaseClassImplementation$Const class CaseClassImplementation$App) = 7
+
+ArrayImplementation
+underlying rep: class [Ljava.lang.Object;
+1
+test1 OK
+1 = 1
+test2 OK
+1 = 1
+
+underlying rep: class [Ljava.lang.Object;
+7
+test3 OK
+AppliedOp([Ljava.lang.Object;@5a39699c class [Ljava.lang.Object; class [Ljava.lang.Object;) = 7
```

Comment:
Ah, the test fail because this address won't match (see http://dotty-ci.epfl.ch/lampepfl/dotty/5372/4 line 10323), so don't fully `toString` the thing I guess...

Reply:
I replaced `Array` with `List` to have a clean `toString`